### DISCUSSION TITLE
Using the EDC Connector requires that I make my own launcher. What are the reasons?

### DISCUSSION THREAD

**Discussion by fhennig** [eclipse-edc/Connector#3263] [Q&A] [answered]
Hi,

I'm currently working with the EDC Connector and as the README states, I have to build my own launcher. An example is provided, but there are no released jars and it is stated explicitly that this is intentional. 

I wanted to ask open-ended about the reasons for this, as I think it would be great to have a versioned, released connector. I think I ready-to-use jar would be great for adoption and also I could be certain that what I'm using is the right thing, it's one less thing that can go wrong. 

Is this only during this early period of the project, or is this a policy that will be kept also as the project matures? 

Cheers

**Comment by juliapampus** [comment, marked as answer]
Thx for the question! The Eclipse Dataspace Components is a framework with a lot of basic functionality for interacting in a dataspace, but also technology-specific features. As the project team does not know your use case and your requirements, we cannot release a ready-to-use bundle of components. There will always be a need to customize the implementation and the deployment. This is why we will, beyond the initial phase of this project, not provide such artifacts like an executable .jar. You can find the project's artifacts on [Maven Central](https://central.sonatype.com/search?q=org.eclipse.edc). 

You may compare it with Spring: You decide to develop a software using Spring. This comes with basic functionalities implementing standards, but it does not provide a ready-to-use .jar file. Because no one knows what software you want to implement. Same goes for the EDC: You decide to develop a dataspace component, you can use the EDC for this. It provides basic functionalities implementing and complying with standards. However, it cannot provide you with the "safety to do everything right". Defining how a dataspace connector should look like, to what IAM it should connect, what vocabulary it should support, whether to use federated cataloging or a centralized catalog service is in the responsibility of the dataspace (project). In addition, usually, there are a lot of concrete requirements of the IT landscape in which the connector is to be deployed. Although with the help of the EDC, you actually need to develop the software to make it fit your use case or the specifications to be met. 

It's the dataspace projects' or operating initiatives' responsibility (or chance) to provide ready-to-use dataspace components (in form of products) that you may use for your use case.

**Reply by fhennig** [reply]
Thanks for your quick response!
